bfsdk/include/bftypes.h

```diff
@@ -84,8 +84,12 @@
 
 #ifdef ENABLE_BUILD_TEST
 #define VIRTUAL virtual
+#define PURE = 0
+#define OVERRIDE override
```

Comment:
I'm assuming that this ```#define VIRTUAL``` macro exists as a performance optimization to avoid the overhead of a v-table. For the vcpu interface definitions, I really like the rigid compiler checks that can be accomplished by combining pure virtual functions with the C++11 ```override``` keyword, so what are everyone's thoughts about providing similar macros to accomplish this? For an example of how these new macros are used, see [here](https://github.com/JaredWright/hypervisor/blob/af2d5e7187bc68e2c1c989ba88dfd8c5b2372f19/bfvmm/include/interface/arch/intel_x64/cpuid.h#L49) and [here](https://github.com/JaredWright/hypervisor/blob/af2d5e7187bc68e2c1c989ba88dfd8c5b2372f19/bfvmm/include/hve/arch/intel_x64/vcpu.h#L367)

Reply:
I would remove this change as it is not needed. The VIRTUAL macro is only needed for unit testing. The difference between `VIRTUAL` and `virtual` is that VIRTUAL is used to state that the function can be mocked, but not actually overriden. If it can be overridden, mark it with `virtual` and then any overrides need to be marked with `override`. This is a simple fix for a known issue in C++ with respect to mocking. PURE and OVERRIDE are not needed because those would only be needed if you intend to use the `virtual` version.

Reply:
also, see comments below